### DISCUSSION TITLE
Community.cooperative_tradeoff interpretation

### DISCUSSION THREAD

**Discussion by WackerO** [micom-dev/micom#23] [Q&A] [answered]
Hello,

I am working with a micom.community model of two bacteria and not quite sure how to understand the output of the tradeoff function:

SA_SE_community.cooperative_tradeoff(fraction=0.8, fluxes=True, pfba=True)

I accessed the fluxes and displayed them and the dataframe looks like this:

![tradeoff](https://user-images.githubusercontent.com/43847497/109523904-2b0c6f00-7ab0-11eb-840f-2e649da8a607.PNG)

Does the general rule hold that negative fluxes represent import, i.e. S. epidermidis consumes for example alanine?
Why does S. epidermidis use separate fluxes? I looked up the compartments of the single models and the community and they look like this:
```
Community compartments:
{'c__iYS854': '', 'e__iYS854': '', 'w__iYS854': '', 'm': '', 'p__Staphylococcus_epidermidis_ATCC_12228': '',
'c__Staphylococcus_epidermidis_ATCC_12228': '', 'e__Staphylococcus_epidermidis_ATCC_12228': ''}
S. aureus compartments:
{'c': 'cytosol', 'w': 'wildtype staph aureus', 'e': 'extracellular space'}
S epidermidis compartments:
{'c': 'cytosol', 'e': 'extracellular space', 'p': 'periplasm'}
```
I would have assumed that both organisms use the community compartment m, but for some reason, this is only used by S. aureus and the medium. Conversely, e.g. in EX_cl_e, both organisms have fluxes, whereas the medium uses the _m flux.
Also, how does it make sense that for EX_cl_e, all three fluxes are negative? If the medium flux is negative, does this mean the medium is "importing", i.e. both organisms are secreting a flux? This would not work here because the organisms both require the metabolite.

Would be great if someone could help me understand this :)

**Comment by cdiener** [comment, marked as answer]
Hi,

You are right that the default direction for exchanges in AGORA is export, so negative fluxes denote imports. This doesn't have to be true in general though. You can check by looking at `community.reactions.EX_cl_e__iYS854.reaction` for instance to verify.

NaN means the reaction is absent in this organism/compartment.

It's a bit hard to tell but the exchanges ending in `_e` are exchanges for individual bacteria, whereas exchanges ending in `_m` are imports and exports in the environment or medium. So those are never associated with any taxon. So for instance for alanine, the molecule is entering the extracellular space at a flux of 2.28. All of this is imported by *S. epidermis*. You see a higher flux for *S. epidermis* because fluxes for individual taxa are scaled to 1 gDW. In your system you have 0.5 gDW of *S. epidermis* / gDW of total biomass, so the flux is twice as high.

For chlorine having all fluxes negative just means that it enters the system at a particular flux and is imported by both taxa, so that makes sense.

FWIW I also think this format for fluxes is a bit hard to understand. My plan is to change it to something more like the `grow` workflow returns.

**Reply by WackerO** [reply]
Thanks very much, great explanation!
